src/main.rs

```diff
@@ -54,11 +63,47 @@ struct Cli {
 
     #[command(flatten)]
     verbose: Verbosity,
+
+    /// Path to the configuration file.
+    #[arg(long, short)]
+    config: Option<String>,
 }
 
 pub async fn inner() -> anyhow::Result<()> {
     let cli = Cli::parse();
 
+    // Start a new Figment instance with default values
+    let mut figment =
+        Figment::new().admerge(Serialized::from(config::Config::default(), "default"));
+
+    // If provided, check config file from environment
+    if let Ok(config_file) = env::var("SPROCKET_CONFIG") {
+        figment = figment.admerge(Toml::file(config_file));
+    }
+
+    // Check XDG_CONFIG_HOME for a config file
+    if let Ok(xdg_config_home) = env::var("XDG_CONFIG_HOME") {
+        figment = figment.admerge(Toml::file(
+            Path::new(&xdg_config_home.as_str()).join("sprocket.toml"),
+        ));
+    }
+
+    // Check HOME for a config file
+    if let Some(home) = home_dir() {
+        figment = figment.admerge(Toml::file(home.join(".config").join("sprocket.toml")));
+    }
```

Comment:
I don't think we need both of these, but I don't know which one we should keep.

Reply:
Looking at this further, the [spec](https://specifications.freedesktop.org/basedir-spec/latest/) says the following:

> $XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

So I'll plan to combine these so that `XDG_CONFIG_HOME` is preferred, but if it is unset, then `$HOME/.config` will be used.